src/shared/sam/cli/samCliLocator.ts

```diff
@@ -69,8 +71,29 @@ abstract class BaseSamCliLocator {
 
         for (const fullPath of fullPaths) {
             this.logger.verbose(`Searching for SAM CLI in: ${fullPath}`)
+            const context: SamCliValidatorContext = {
+                samCliLocation: async () => fullPath,
+                getSamCliExecutableId: async () => 'temp',
```

Comment:
If "temp" shows up in a log or error it will be confusing. I would name this something that more obviously signals "something is wrong".
```suggestion
                getSamCliExecutableId: async () => 'bogus',
```